sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -871,34 +850,26 @@ public void setSpanContext(
       final @NotNull Throwable throwable,
       final @NotNull ISpan span,
       final @NotNull String transactionName) {
-    Objects.requireNonNull(throwable, "throwable is required");
-    Objects.requireNonNull(span, "span is required");
-    Objects.requireNonNull(transactionName, "transactionName is required");
-    // to match any cause, span context is always attached to the root cause of the exception
-    final Throwable rootCause = ExceptionUtils.findRootCause(throwable);
-    // the most inner span should be assigned to a throwable
-    if (!throwableToSpan.containsKey(rootCause)) {
-      throwableToSpan.put(rootCause, new Pair<>(new WeakReference<>(span), transactionName));
-    }
-  }
-
-  // TODO this seems unused
-  @Nullable
-  SpanContext getSpanContext(final @NotNull Throwable throwable) {
-    Objects.requireNonNull(throwable, "throwable is required");
-    final Throwable rootCause = ExceptionUtils.findRootCause(throwable);
-    final Pair<WeakReference<ISpan>, String> pair = this.throwableToSpan.get(rootCause);
-    if (pair != null) {
-      final WeakReference<ISpan> spanWeakRef = pair.getFirst();
-      if (spanWeakRef != null) {
-        final ISpan span = spanWeakRef.get();
-        if (span != null) {
-          return span.getSpanContext();
-        }
-      }
-    }
-    return null;
-  }
+    Sentry.getGlobalScope().setSpanContext(throwable, span, transactionName);
+  }
+
+  //  // TODO this seems unused
```

Comment:
Remove if still present in follow up PRs

Reply:
Already removed in a follow up PR